cardano-config/test/Test/Cardano/Config/Gen.hs

```diff
@@ -81,13 +83,62 @@ genShelleyGenesis =
     <*> Gen.word64 (Range.linear 1 100000)
     <*> fmap fromIntegral (Gen.word $ Range.linear 0 100000)
     <*> Gen.word64 (Range.linear 1 100000)
-    <*> fmap fromIntegral (Gen.word $ Range.linear 100 1000000)
-    <*> fmap fromIntegral (Gen.word $ Range.linear 100 1000000)
+    <*> genPParams
     <*> fmap Map.fromList genGenesisDelegationList
     <*> fmap Map.fromList genFundsList
     <*> pure emptyGenesisStaking
 
 
+genPParams :: Gen PParams
+genPParams =
+  PParams
+    <$> genNatural (Range.linear 0 1000)
+    <*> genNatural (Range.linear 0 3)
+    <*> fmap fromIntegral (Gen.word $ Range.linear 100 1000000)
+    <*> fmap fromIntegral (Gen.word $ Range.linear 100 1000000)
+    <*> fmap fromIntegral (Gen.word $ Range.linear 100 1000000)
+    <*> genCoin
+    <*> genUnitInterval
+    <*> genRational
+    <*> genCoin
+    <*> genUnitInterval
+    <*> genRational
+    <*> genEpochNo
+    <*> genNatural (Range.linear 0 10)
+    <*> genRational
+    <*> genUnitInterval
+    <*> genUnitInterval
+    <*> genUnitInterval
+    <*> genNonce
+    <*> genProtVer
+
+genNatural :: Range Natural -> Gen Natural
+genNatural range = Gen.integral range
+
+genRational :: Gen Rational
+genRational = Gen.realFrac_ (Range.linearFrac 0 10000)
+
+genEpochNo :: Gen EpochNo
+genEpochNo = EpochNo <$> Gen.word64 (Range.linear 0 500)
+
+genNonce :: Gen Nonce
+genNonce =
+  Gen.choice
+    [ mkNonce <$> genNatural (Range.linear 1 123)
+    , pure NeutralNonce
+    ]
+
+genProtVer :: Gen ProtVer
+genProtVer =
+  ProtVer
+    <$> genNatural (Range.linear 0 1000)
+    <*> genNatural (Range.linear 0 1000)
+
+genUnitInterval :: Gen UnitInterval
+genUnitInterval =
+  UnsafeUnitInterval
+    <$> Gen.realFrac_ (Range.linearFrac 0 1)
```

Comment:
According to the hedgehog docs `realFrac_` does not shrink. Should we use the `double` generator here instead?

Reply:
Shrinking not too important here.